lib/backend.js

```diff
@@ -391,7 +391,8 @@ Backend.prototype.fetch = function(agent, index, id, options, callback) {
     collection: collection,
     id: id
   };
-  var snapshotOptions = options && options.snapshotOptions;
+  var snapshotOptions = (options && options.snapshotOptions) || {};
+  if (agent) snapshotOptions.custom = agent.custom;
```

Comment:
How about `snapshotOptions.agentCustom` to make it clear where it's coming from? `options.custom` is very general really give a hint about where it's coming from.